src/legacy/core_plugins/metrics/public/components/vis_editor.js

```diff
@@ -59,6 +61,10 @@ class VisEditor extends Component {
     this.props.vis.uiStateVal(field, value);
   };
 
+  updateVisState = debounce(() => {
```

Comment:
Adding debounce here means that we could be calling `this.props.vis.updateState()` after the component has been unmounted, can we protect against that?

Reply:
thanks